cfme/utils/appliance/implementations/ui.py

```diff
@@ -340,9 +339,10 @@ def check_for_badness(self, fn, _tries, nav_args, *args, **kwargs):
                     br.widgetastic.is_displayed(".modal-backdrop.fade.in")):
                 logger.warning("Page was blocked with blocker div, recycling.")
                 recycle = True
-            elif cfme_exc.is_cfme_exception():
+            elif br.widgetastic.is_displayed("//div[@id='exception_div']"):
                 logger.exception("CFME Exception before force navigate started!: {}".format(
-                    cfme_exc.cfme_exception_text()))
+                    br.widgetastic.text(
+                        "//div[@id='exception_div']//td[@id='maincol']/div[2]/h3[2]")))
```

Comment:
**QUESTION**
@mfalesni I'm not sure if it correct usage `is_displayed()` and `text()` here.

Reply:
It seems okay.